nautobot_ssot/integrations/bootstrap/signals.py

```diff
@@ -83,13 +78,21 @@ def nautobot_database_ready_callback(sender, *, apps, **kwargs):  # pylint: disa
     }
 
     if LIFECYCLE_MGMT:
-        signal_to_model_mapping.update(
-            {
-                "software": SoftwareLCM,
-                "software_image": SoftwareImageLCM,
-                "validated_software": ValidatedSoftwareLCM,
-            }
-        )
+        try:
+            SoftwareLCM = apps.get_model("nautobot_device_lifecycle_mgmt", "SoftwareLCM")
+            signal_to_model_mapping["software"] = SoftwareLCM
+        except LookupError as err:
+            print(f"Unable to find SoftwareLCM model from Device Lifecycle Management App. {err}")
```

Comment:
In my mind we should either.

- Rely on pyproject.toml to ensure we side step this issue by pinning / setting the proper versions of dependencies
- Support both new and old versions

The print to me would not be easily seen, and as far as I can tell, not let the customer know the issue is exactly.

Reply:
@itdependsnetworks The problem is this isn't a direct dependency. The Bootstrap integration supports both Golden Config and Device Lifecycle Management. The problem is that with DLM's change to get rid of SoftwareLCM and SoftwareImageLCM we need to account for that with this integration as it was blowing up my testing of the DLM update.